Sources/NIO/Bootstrap.swift

```diff
@@ -364,8 +385,121 @@ public final class ServerBootstrap {
+                case .backlog(let value):
+                    _ = serverBootstrap.serverChannelOption(ChannelOptions.backlog, value: value)
+                }
+            }
+        }
+    }
+    
+    /// A channel option which can be applied to bootstrap using shorthand notation.
+    /// - See: ServerBootstrap.serverOptions(_ options: [ShorthandBootstrapOption])
+    public struct ShorthandChildBootstrapOption {
+        private let data: ShorthandChildOption
+        
+        private init(_ data: ShorthandChildOption) {
+            self.data = data
+        }
+        
+        /// Apply the contained option to the supplied ServerBootstrap
+        /// - Parameter serverBootstrap: bootstrap to apply this option to.
+        /// - Returns: the modified bootstrap (currently the same one mutated)
+        func applyOption(to serverBootstrap: ServerBootstrap) {
+            data.applyOption(to: serverBootstrap)
+        }
+        
+        fileprivate enum ShorthandChildOption {
+            case reuseAddr(Bool)
+            case allowRemoteHalfClosure(Bool)
+            
+            func applyOption(to serverBootstrap: ServerBootstrap) {
+                switch self {
+                case .reuseAddr(let value):
+                    _ = serverBootstrap.childChannelOption(ChannelOptions.socketOption(.reuseaddr),
+                                                           value: value ? 1 : 0)
+                case .allowRemoteHalfClosure(let value):
+                    _ = serverBootstrap.childChannelOption(ChannelOptions.allowRemoteHalfClosure, value: value)
+                }
+            }
+        }
+    }
+}
+
+// Hashable for the convenience of users.
+extension ServerBootstrap.ShorthandServerBootstrapOption : Hashable {}
+extension ServerBootstrap.ShorthandServerBootstrapOption.ShorthandServerOption : Hashable {}
+
+/// Approved shorthand server options.
+extension ServerBootstrap.ShorthandServerBootstrapOption {
+    /// Option to reuse address.
+    /// - See:  NIOBSDSocket.Option.reuseaddr
+    public static let reuseAddr = ServerBootstrap.ShorthandServerBootstrapOption(.reuseAddr(true))
+    
+    /// Option to disable autoRead
+    /// - See: ChannelOptions.autoRead
+    public static let disableAutoRead = ServerBootstrap.ShorthandServerBootstrapOption(.autoRead(false))
+    
+    /// `BacklogOption` allows users to configure the `backlog` value as specified in `man 2 listen`.
+    /// This is only useful for `ServerSocketChannel`s.
+    /// - See: ChannelOptions.backlog
+    public static func backlog(_ value: ChannelOptions.Types.BacklogOption.Value) ->
+        ServerBootstrap.ShorthandServerBootstrapOption {
+        return ServerBootstrap.ShorthandServerBootstrapOption(.backlog(value))
+    }
+}
+
+// Hashable for the convenience of users.
+extension ServerBootstrap.ShorthandChildBootstrapOption : Hashable {}
+extension ServerBootstrap.ShorthandChildBootstrapOption.ShorthandChildOption : Hashable {}
```

Comment:
Nit: As hashable is (in this case) courtesy of the compiler, and we own the definitions, I'd probably just move the washable conformance to the root definition. Feel free to ignore...

Reply:
Going to ignore this one for now.